doc/api/esm.md

```diff
@@ -1160,25 +1160,26 @@ condition list **must** be passed through to the `defaultResolve` function.
 ```js
 /**
  * @param {string} specifier
- * @param {object} context
- * @param {string} context.parentURL
- * @param {string[]} context.conditions
- * @param {function} defaultResolve
- * @returns {object} response
- * @returns {string} response.url
+ * @param {{
+ *   parentURL:!(string|undefined),
+ *   conditions:!(Array<string>),
+ * }} context
+ * @param {Function} defaultResolve
+ * @returns {{url:string}}
  */
-export async function resolve(specifier, context, defaultResolve) {
+export function resolve(specifier, context, defaultResolve) {
```

Comment:
shouldn’t this stay async?

Reply:
Not sure what your thoughts are on [the `require-await` lint rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/require-await.md) or if this is an exception to that rule.

Reply:
That rule is an abomination and should never have been accepted into eslint core, and should never be used by anyone. `async function` is hugely useful by itself to statically ensure a promise is returned, and thrown exceptions reject it. `await` is never needed.

Reply:
Lol, sounds good, I'll make it async again. :)